### ISSUE TITLE
ndslice: concatenate/vstack/hstack [$15 awarded]

### ISSUE THREAD

**Issue by wilzbach** [9il/dip80-ndslice#213]
Idea: Join multiple ndslices to a bigger ndslice
Maybe sth. like:

``` D
auto a = iotaSlice(2, 4);
auto b = 2.repeat.sliced(2, 4);
a.concatenate(b);
```

(Forum [discussion](http://forum.dlang.org/post/goxmvgwcsbzavrdpzjjn@forum.dlang.org))
##

**Comment by 9il** [comment]
~~next week plan:~~
- [ ] `selection.elementsMap` - shell with `opIndex`, `length` and (optional) `save` methods.
- [ ] `selection.concatenated` - concatenation without allocation based on `elementsMap`
- [ ] `slice.elements` /  `slice.makeElements` - allocates a common array composed of slice elements.
- [ ] `slice.concatenation` / ``slice.makeConcatenation` - multidimensional concatenation with allocation

**Comment by wilzbach** [comment]
I like the Api

1) probably a good opportunity to think more about a public isSlice method? 
2) We shouldn't forgot there are multiple ways to stack (see below the way NumPy does it), but maybe we can have this as template argument? 

http://docs.scipy.org/doc/numpy-1.10.1/reference/generated/numpy.vstack.html
http://docs.scipy.org/doc/numpy-1.10.1/reference/generated/numpy.concatenate.html#numpy.concatenate
http://docs.scipy.org/doc/numpy-1.10.1/reference/generated/numpy.stack.html#numpy.stack
http://docs.scipy.org/doc/numpy-1.10.1/reference/generated/numpy.hstack.html#numpy.hstack
http://docs.scipy.org/doc/numpy-1.10.1/reference/generated/numpy.dstack.html#numpy.dstack

**Comment by 9il** [comment]
> but maybe we can have this as template argument

yes

**Comment by Marenz** [comment]
Seeing that this was created May 15th and reading "next week plan" leads me to the suspicion that the issue might have been forgotten. So I figure a minor ping might be in order. _ping_

**Comment by 9il** [comment]
@Marenz GLAS and ndslice.algorithm just have higher priority.

**Comment by Marenz** [comment]
Sure, no pressure, I just pinged in case it was forgotten ;)
Thanks for the work!

**Comment by 9il** [comment]
@Marenz Thank you for the first Mir bounty! Currently we have issue with bountyscore, because Mir project is registered for [DlangScience](https://www.bountysource.com/teams/dlangscience) instead of [Mir](https://www.bountysource.com/teams/mir) team. I hope we will fix it #345 soon

**Comment by 9il** [comment]
Bounty is located in https://www.bountysource.com/teams/libmir now. I have sent query to get access for this bountysource team.
